kserve/hack/update-kserve-manifests.sh

```diff
@@ -0,0 +1,20 @@
+#!/usr/bin/env bash
+
+echo "Updating KServe manifests"
+script_dir=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
+kserve_dir="$script_dir/../kserve"
+tmp_dir="$(mktemp -d)"
+
+git init "$tmp_dir"
+cd "$tmp_dir" || exit
+git remote add -f origin https://github.com/kserve/kserve.git
+git config core.sparseCheckout true
+echo "config" >> .git/info/sparse-checkout
+git pull origin master
+
+rm -rf "$kserve_dir"
+mkdir "$kserve_dir"
+
+mv ./config/* "$kserve_dir"
```

Comment:
Can we refer to the github url rather than doing a hard-copy?
For example, in `kserve/odh-overlays/controller/kustomization.yaml`, rather than having:
```yaml
resources:
  - ../../kserve/default/
```

you could have:
```yaml
resources:
  - github.com/opendatahub-io/kserve/config/default?ref=master
```

(note, I didn't test this, but this docs say it should be possible: https://kubectl.docs.kubernetes.io/references/kustomize/kustomization/resource/)

I don't like too much copying/synching files.

Reply:
Awesome! I did not know that, now we can drop 99% of the stuff in this PR :)

Reply:
I don't think this is a good idea.. because upstream kserve/kserve manifests can be changed anytime and definitely it will impact to OpenDataHub. We need to have a stable manifests always.

Reply:
I think `ref=master` could be replaced by `ref={commitHash}` to make it stable.

I also saw that it is changed to use `opendatahub-io/kserve` rather than upstream. So, upstream becomes less of a concern.

Reply:
That looks better.

Reply:
I'll do the updates, depending on the outcome of the discussion in https://github.com/opendatahub-io/odh-manifests/pull/838#discussion_r1227565608

Reply:
Yeah, that would be resolved if we do https://github.com/opendatahub-io/odh-manifests/pull/838#discussion_r1227565608. Then we'd have the final YAMLs in this repo here. Would that work?

Reply:
> Then we'd have the final YAMLs in this repo here. Would that work?

That works and I think it also would work with the outdated kfdef, because we would need minimal parametrs for the operator.